## This notebook will contain the code for problems 3,4,9, and 10

First Data and package loading

In [ ]:
import numpy as np
from scipy import stats
import matplotlib.pyplot as plt
import pandas as pd

In [ ]:
InitData = pd.read_csv("movieReplicationSet.csv")

### Problem 3: We will test if ratings of shrek are gendered with a Mann-Whitney U Test

In [ ]:
#So we need to create a shrek and gender database
Quest3 = InitData[["Shrek (2001)", "Gender identity (1 = female; 2 = male; 3 = self-described)"]]
#and clean it
Cle3 = Quest3.dropna()

In [ ]:
#Next break it up into men and women
fem3 = Cle3[Cle3["Gender identity (1 = female; 2 = male; 3 = self-described)"] == 1]
man3 = Cle3[Cle3["Gender identity (1 = female; 2 = male; 3 = self-described)"] == 2]

In [ ]:
#Then run the Mann Whitnety Test
stats.mannwhitneyu(fem3["Shrek (2001)"], man3["Shrek (2001)"])

MannwhitneyuResult(statistic=96830.5, pvalue=0.050536625925559)

And we see this is not significant at p = 0.05

### Problem 4: We will create a function that performs the Mann-Whitney U test from 3 and loop it across all movies

In [ ]:
#Our Function
def quest4(movie):
    Quest4 = InitData[[movie, "Gender identity (1 = female; 2 = male; 3 = self-described)"]]
    Cle4 = Quest4.dropna()
    fem4 = Cle4[Cle4["Gender identity (1 = female; 2 = male; 3 = self-described)"] == 1]
    man4 = Cle4[Cle4["Gender identity (1 = female; 2 = male; 3 = self-described)"] == 2]
    results = stats.mannwhitneyu(fem4[movie], man4[movie])[1]
    return results

In [ ]:
#Our Loop
movies = []
pvalue = []

#I need to grab the column names for all the movies (put them in a listO
for x in range(0,400):
    title = str(InitData.columns[x])
    movies.append(title)
    pvalue.append(quest4(title))

Outputframe = pd.DataFrame({'movies' : movies,'pvalue' : pvalue}, columns = ['movies','pvalue'])

In [ ]:
#Lets find our ratio
print(len(Outputframe[Outputframe['pvalue'] < 0.005])/400)

0.125


In [ ]:
#I'm also exporting the data to look at the values
Sortframe = Outputframe.sort_values(by = 'pvalue')
Sortframe = Sortframe.reset_index(drop = True)
Sortframe.to_csv("Question 4 Sorted.csv")

We find that 12.5% of movies are rated differently between men and women. The movies that make this up like "The Proposal" and "Batman: The Dark Knight" make sense

### Problem 9: We will check the distributions of Home Alone and Finding Nemo with a KS test

In [ ]:
#this will be an element wise drop so I will create two new dataframes
Home = InitData['Home Alone (1990)']
Nemo = InitData['Finding Nemo (2003)']

Home = Home.dropna()
Nemo = Nemo.dropna()

In [ ]:
stats.kstest(Home, Nemo)

KstestResult(statistic=0.15269080020897632, pvalue=6.379397182836346e-10, statistic_location=3.0, statistic_sign=1)

We find the difference in the distributions is significant

In [ ]:
#Histograms if needed
plt.hist(Home, density = True, bins = 8)
plt.show()
plt.hist(Nemo,  density = True, bins = 8)
plt.show()

### Problem 10: We will check the consistency of several movie franchises with a Kruskall Wallis test on each

In [ ]:
#Make a function to run on each movie franchise
def series(word):
    #create frame
    Frame = InitData.filter(like = word)
    Movies = []
    for x in Frame.columns:
        Movies.append(Frame[x].dropna())
    return stats.kruskal(*Movies)

In [ ]:
series('Star Wars')

KruskalResult(statistic=230.5841753686405, pvalue=8.01647736660335e-48)

In [ ]:
series('Harry Potter')

KruskalResult(statistic=3.331230732890868, pvalue=0.3433195083728917)

In [ ]:
series('The Matrix')

KruskalResult(statistic=48.378866521305774, pvalue=3.1236517880781424e-11)

In [ ]:
series('Indiana Jones')

KruskalResult(statistic=45.79416340261569, pvalue=6.272775639796079e-10)

In [ ]:
series('Jurassic Park')

KruskalResult(statistic=46.59088064385298, pvalue=7.636930084362221e-11)

In [ ]:
series('Pirates of the Caribbean')

KruskalResult(statistic=20.64399756002606, pvalue=3.2901287079094474e-05)

In [ ]:
series('Toy Story')

KruskalResult(statistic=24.38599493626327, pvalue=5.065805156537524e-06)

In [ ]:
series('Batman')

KruskalResult(statistic=190.53496872634642, pvalue=4.2252969509030006e-42)

And they are all significant outside of Harry Potter